### PR TITLE
Fix CODEOWNERS format; Add Outside Collaborators

### PR DESCRIPTION
**Pull request by ScriptAutomate**
### What does this PR do?

> Related: https://github.com/github/docs/issues/18984

Fixes `CODEOWNERS` file.

- [Info on `CODEOWNERS`](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners)
- Main problem: Most of the teams listed did not have at least `write` permissions (level required to be included) when it came to this repository, or the team didn't exist.

Problems fixed:
- Following teams don't have proper permissions for repo, and are being excluded
  - `@saltstack/team-boto`
  - `@saltstack/team-cloud`
  - `@saltstack/team-spm`
  - `@salstack/team-ssh`
  - `@saltstack/team-state`
  - `@saltstack/team-suse` (this group is also empty due to conversion of members to Outside Collaborators)
  - `@saltstack/team-transport`
- Following teams do not exist
  - `@saltstack/team-netapi`
  - `@saltstack/team-proxy`
  - `@saltstack/team-code`
- Outside Collaborators were originally added, but are now all removed due to inability for CODEOWNERS to work with members without **Write** access to the repo:
  - @ryan-lane 
  - @verhulstm 
  - @pcn 
  - @nicholasmhughes 
  - @meaksh 
  - @isbm 
  - @agraul 
  - @vzhestkov 
  - @aplanas 
- Removed `.ci/*` as it no longer exists in repo

### Previous Behavior

Broken `CODEOWNERS` file :cry: 

### New Behavior

Functional `CODEOWNERS` file :smile: 

### Commits signed with GPG?
No

### CODE REVIEW

**Comment by ScriptAutomate** [comment]
As a follow-up on CODEOWNERS assignment:

It looks like non-VMware employees will not be able to be assigned as code reviewers. GitHub discovered that their documentation was incorrect, and that a reviewer would need to have Write permissions to the repository for GitHub automation to work as expected. Due to only giving Outside Collaborators Read or Triage permissions, we will not be able to have CODEOWNERS automation for the salt repository that includes community members unless GitHub supports this or we try to find an alternative means.

Original GitHub docs (which were incorrect):

> The people you choose as code owners must have **read permissions** for the repository. When the code owner is a team, that team must be visible and it must have write permissions, even if all the individual members of the team already have write permissions directly, through organization membership, or through another team membership.

New GitHub docs mention:

> The people you choose as code owners must have **write permissions** for the repository. When the code owner is a team, that team must be visible and it must have write permissions, even if all the individual members of the team already have write permissions directly, through organization membership, or through another team membership.

At this point, the CODEOWNERS file will only be able to work with VMware-employees, and not community members. I'll leave this up to the open devs to modify as they see fit with VMware employees when it comes to review assignment.

More info:
- https://github.com/github/docs/issues/18984#issuecomment-1218538679

**Review by s0undt3ch — DISMISSED** [review]

**Comment by ScriptAutomate** [comment]
Related GitHub feature request has been opened:
- [Feature Request: Allow users with Triage and/or Read Permissions to be supported by CODEOWNERS](https://github.com/orgs/community/discussions/29982)

**Review by whytewolf — APPROVED** [review]

**Review by garethgreenaway — APPROVED** [review]

**Pull request merged**
